src/App.tsx

```diff
@@ -1,14 +1,82 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+
+export const FILTER_ALL = 'all';
+export const FILTER_COMPLETE = 'completed';
+export const FILTER_ACTIVE = 'active';
+
+function getFilterTodos(todos:Todo[], filter:string):Todo[] {
+  const preparedTodos = [...todos];
+
+  if (filter === FILTER_ALL) {
+    return preparedTodos;
+  }
+
+  if (filter === FILTER_COMPLETE) {
+    return preparedTodos.filter(todo => todo.completed);
+  }
+
+  if (filter === FILTER_ACTIVE) {
+    return preparedTodos.filter(todo => !todo.completed);
+  }
+
+  return preparedTodos;
+}
+
+function inputFilterTodos(todos:Todo[], query:string) {
+  const normalizedQuery = query.trim().toLowerCase();
+  let copyTodos = [...todos];
+
+  if (normalizedQuery) {
+    copyTodos = copyTodos.filter(
+      todo => todo.title.toLowerCase().includes(normalizedQuery),
+    );
+  }
+
+  return copyTodos;
+}
```

Comment:
Please move this function to utils so as not to clutter the code of your component